src/main/scala/dpla/ingestion3/harvesters/api/CdlHarvester.scala

```diff
@@ -17,9 +21,66 @@ import scala.util.{Failure, Success, Try}
   * Calisphere API documentation
   * https://help.oac.cdlib.org/support/solutions/articles/9000101639-calisphere-api
   *
-  * @param queryParams - Query parameters (q, rows, cursorMark etc.)
   */
-class CdlHarvester(val queryParams: Map[String, String] = Map()) {
+class CdlHarvester(conf: i3Conf, outputDir: String, harvestLogger: Logger)
+  extends ApiHarvester(conf, outputDir, harvestLogger) {
+
+  override protected val mimeType: String = "application_json"
+
+  override protected val queryParams: Map[String, String] = Map(
+    "query" -> conf.harvest.query,
+    "rows" -> conf.harvest.rows,
+    "api_key" -> conf.harvest.apiKey
+  ).collect{ case (key, Some(value)) => key -> value } // remove None values
+
+  override protected def localApiHarvest: Unit = {
+    harvestLogger.info(s"Query params: ${queryParams.toString}")
+
+    implicit val formats = DefaultFormats
+
+    // Mutable vars for controlling harvest loop
+    var continueHarvest = true
+    var cursorMark = "*"
+
+    // Runtime tracking
+    val startTime = System.currentTimeMillis()
+
+    while(continueHarvest) getSinglePage(cursorMark) match {
+      case error: ApiError with ApiResponse =>
+        harvestLogger.error("Error returned by request %s\n%s\n%s".format(
+          error.errorSource.url.getOrElse("Undefined url"),
+          error.errorSource.queryParams,
+          error.message
+        ))
+        continueHarvest = false
+      case src: ApiSource with ApiResponse =>
+        src.text match {
+          case Some(docs) =>
+            val json = parse(docs)
+            val cdlRecords = (json \\ "docs").children.map(doc => {
+              ApiRecord((doc \\ "identifier").toString, compact(render(doc)))
+            })
+
+            // @see ApiHarvester
+            saveOut(cdlRecords)
+
+            // Loop control
+            cursorMark = (json \\ "cursorMark").extract[String]
+            val nextCursorMark = (json \\ "nextCursorMark").extract[String]
+
+            cursorMark.matches(nextCursorMark) match {
+              case true => continueHarvest = false
+              case false => cursorMark = nextCursorMark
+            }
+          case None =>
+            harvestLogger.error(s"The body of the response is empty. Stopping run.\nCdlSource >> ${src.toString}")
+            continueHarvest = false
+        }
+      case _ =>
+        harvestLogger.error("Harvest returned None")
+        continueHarvest = false
+    }
+  }
 
   /**
     * Get a single-page, un-parsed response from the OAI feed, or an error if
```

Comment:
Noticing my own mistake here: `CDL` not `OAI` feed

Reply:
Good catch!